### PR TITLE
add tests in pkg/transport package

### PR DESCRIPTION
**Pull request by yichengq**
for #1467

### CODE REVIEW

File: pkg/transport/keepalive_listener_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package transport
+
+import (
+	"net/http"
+	"testing"
+)
+
+// TestNewKeepAliveListener tests NewKeepAliveListener returns a listener
+// that accepts connections.
+func TestNewKeepAliveListener(t *testing.T) {
```
Comment (xiang90) [inline pkg/transport/keepalive_listener_test.go]: Add todo, we still need to verify the keepalive option is set correctly.

File: pkg/transport/listener_test.go [hunk 2]
```diff
@@ -44,6 +45,41 @@ func fakeCertificateParserFunc(cert tls.Certificate, err error) func(certPEMBloc
 	}
 }
 
+// TestNewListenerTLSInfo tests that NewListener with valid TLSInfo returns
+// a TLS listerner that accepts TLS connections.
+func TestNewListenerTLSInfo(t *testing.T) {
+	tmp, err := createTempFile([]byte("XXX"))
+	if err != nil {
+		t.Fatalf("Unable to prepare tmpfile: %v", err)
```
Comment (xiang90) [inline pkg/transport/listener_test.go]: unable to create tmpfile. Can you please try to make everything consistent?

File: pkg/transport/listener_test.go [hunk 3]
```diff
@@ -44,6 +45,41 @@ func fakeCertificateParserFunc(cert tls.Certificate, err error) func(certPEMBloc
...
+	defer ln.Close()
+
+	go http.Get("https://" + ln.Addr().String())
+	conn, err := ln.Accept()
+	if err != nil {
+		t.Fatalf("unexpected Accept error: %v", err)
+	}
+	defer conn.Close()
+	if _, ok := conn.(*tls.Conn); !ok {
+		t.Errorf("expected to get *tls.Conn, but failed")
```
Comment (xiang90) [inline pkg/transport/listener_test.go]: failed to get *tls.Conn

File: pkg/transport/listener_test.go [hunk 4]
```diff
@@ -44,6 +45,41 @@ func fakeCertificateParserFunc(cert tls.Certificate, err error) func(certPEMBloc
...
+	}
+}
+
+func TestNewListenerBadTLSInfo(t *testing.T) {
+	tlsInfo := TLSInfo{CertFile: "@badname", KeyFile: "@badname"}
+	_, err := NewListener(":0", "https", tlsInfo)
+	if err == nil {
```
Comment (xiang90) [inline pkg/transport/listener_test.go]: can we compare with the expected error rather than nil?

File: pkg/transport/timeout_transport_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	defer srv.Close()
+	conn, err := tr.Dial("tcp", srv.Listener.Addr().String())
+	if err != nil {
+		t.Fatalf("unexpected dial error: %v", err)
+	}
+	defer conn.Close()
+
+	tconn, ok := conn.(*timeoutConn)
+	if !ok {
+		t.Fatalf("unexpected failure on casting conn to timeoutConn")
```
Comment (xiang90) [inline pkg/transport/timeout_transport_test.go]: Please make this consistent with L70 pkg/transport/listener_test.go

**Comment by xiang90** [comment]
lgtm

**Comment by yichengq** [comment]
issues all resolved.

**Pull request merged**